### ISSUE TITLE
Extension to interact with the public Scratch API

### ISSUE THREAD

**Issue by GarboMuffin** [TurboWarp/extensions#211]
> Could you add an extension that connects to the scratch api? Because all the other methods are complicated and the turbowarp extensions dont work with it. I would code it myself but I have no idea how.

**Comment by DNin01** [comment]
Do they mean things like getting how many views a project has?

**Comment by GarboMuffin** [comment]
I assume so

For anyone that wants to do this, you may want to either use our trampoline (https://trampoline.turbowarp.org/proxy/projects/104) or ScratchDB

**Comment by Mickmick2012** [comment]
i am currently working on a prototype. I will see if it works

**Comment by Dy-tech** [comment]
@GarboMuffin, aren't you talking about S-Grab?

**Comment by LukeManiaStudioss** [comment]
> @GarboMuffin, aren't you talking about S-Grab?

S-Grab currently doesn't connect to the public Scratch API and doesn't even work for some users.

**Comment by Dy-tech** [comment]
> > @GarboMuffin, aren't you talking about S-Grab?
> 
> S-Grab currently doesn't connect to the public Scratch API and doesn't even work for some users.

What does it do then?

**Comment by SurvExe1Pc** [comment]
yeah a few days ago it was not working and now it is?

**Comment by SurvExe1Pc** [comment]
if scratchdb is not working it should at least try to access the proxy for turbowarp, https://trampoline.turbowarp.org/

**Comment by Flappy25** [comment]
> > @GarboMuffin, aren't you talking about S-Grab?
> 
> S-Grab currently doesn't connect to the public Scratch API and doesn't even work for some users.

Can't we just use GET to access data from api.scratch.mit.edu?

**Comment by CST1229** [comment]
> Can't we just use GET to access data from api.scratch.mit.edu?

No, because the Scratch API does not let other sites access it. (TurboWarp uses [its own API proxy](https://trampoline.turbowarp.org) to access it.)

**Comment by pumpkinhasapatch** [comment]
PR #803 uses TurboWarp's trampoline proxy instead of ScratchDB to fetch stuff from the Scratch API.

**Comment by GarboMuffin** [comment]
not planning more than we have today

**Comment by Brackets-Coder** [comment]
wasn't the Scratch API closed/restricted or was that ScratchDB
I thought I remembered them doing something that stopped third party extensions/services from working

**Comment by GarboMuffin** [comment]
some of s-grab is hidden because of that website going down, other parts were migrated over to trampoline

**Comment by GarboMuffin** [comment]
trampoline is kind of always hanging on by a thread and other ways to interact with scratch are not necessarily much better. hence, not planning anything in this area
